feature-libs/user/account/components/otp-login-form/otp-login-form.component.ts

```diff
@@ -0,0 +1,109 @@
+  UntypedFormGroup,
+  Validators,
+} from '@angular/forms';
+import { RoutingService, WindowRef } from '@spartacus/core';
+import { CustomFormValidators } from '@spartacus/storefront';
+import { BehaviorSubject, Observable, tap } from 'rxjs';
+
+import {
+  VerificationToken,
+  VerificationTokenCreation,
+  VerificationTokenFacade,
+} from '@spartacus/user/account/root';
+import { ONE_TIME_PASSWORD_LOGIN_PURPOSE } from '../user-account-constants';
+
+@Component({
+  selector: 'cx-otp-login-form',
+  templateUrl: './otp-login-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OneTimePasswordLoginFormComponent {
+  constructor(
+    protected routingService: RoutingService,
+    protected verificationTokenFacade: VerificationTokenFacade,
+    protected winRef: WindowRef
+  ) {}
+
+  protected busy$ = new BehaviorSubject(false);
+
+  isUpdating$: Observable<boolean> = this.busy$.pipe(
+    tap((state) => {
+      const userId = this.winRef.nativeWindow?.history?.state?.['newUid'];
+      if (userId) {
+        this.form.patchValue({ userId });
+      }
+      state === true ? this.form.disable() : this.form.enable();
+    })
+  );
+
+  form: UntypedFormGroup = new UntypedFormGroup({
+    userId: new UntypedFormControl('', [
+      Validators.required,
+      CustomFormValidators.emailValidator,
+    ]),
+    password: new UntypedFormControl('', Validators.required),
+  });
+
+  @HostBinding('class.user-form') style = true;
+
+  onSubmit(): void {
+    if (!this.form.valid) {
+      this.form.markAllAsTouched();
+      return;
+    }
+
+    this.busy$.next(true);
+    const verificationTokenCreation = this.collectDataFromLoginForm();
+    this.verificationTokenFacade
+      .createVerificationToken(verificationTokenCreation)
+      .subscribe({
+        next: (result: VerificationToken) =>
+          this.goToVerificationTokenForm(result, verificationTokenCreation),
+        error: () => this.busy$.next(false),
+        complete: () => this.onCreateVerificationTokenComplete(),
+      });
+  }
+
+  protected goToVerificationTokenForm(
+    verificationToken: VerificationToken,
+    verificationTokenCreation: VerificationTokenCreation
+  ): void {
+    this.routingService.go(
+      {
+        cxRoute: 'verifyToken',
+      },
+      {
+        state: {
+          loginId: verificationTokenCreation.loginId,
+          password: verificationTokenCreation.password,
+          tokenId: verificationToken.tokenId,
+          expiresIn: verificationToken.expiresIn,
+        },
+      }
+    );
+  }
+
+  protected onCreateVerificationTokenComplete(): void {
+    this.form.reset();
+    this.busy$.next(false);
+  }
+
+  protected collectDataFromLoginForm(): VerificationTokenCreation {
+    return {
+      // TODO: consider dropping toLowerCase as this should not be part of the UI,
+      // as it's too opinionated and doesn't work with other AUTH services
+      loginId: this.form.value.userId.toLowerCase(),
```

Comment:
What will happen if we remove toLowerCase now?

Reply:
this loginId is email, by default in database is lowercase

Reply:
It's better let backend(OCC API) to handle it?

Reply:
maybe here is some strange resaon, I only keep same with old code: https://github.com/SAP/spartacus/blob/8c27c8a72b81535e02694a98a283fe98688a1265/feature-libs/user/account/components/login-form/login-form-component.service.ts#L63